sonarts-core/src/rules/noUnusedArrayRule.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { firstAncestor, is, COMPOUND_ASSIGNMENTS } from "../utils/navigation";
+import { SymbolTableBuilder } from "../symbols/builder";
+import { Usage, SymbolTable, UsageFlag } from "../symbols/table";
+
+export class Rule extends tslint.Rules.TypedRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-unused-array",
+    description: "Array contents should be used",
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-4030",
+    type: "maintainability",
+    typescriptOnly: false,
+  };
+
+  private static writeArrayPatterns: ((statement: ts.ExpressionStatement, usage: Usage) => boolean)[] = [
```

Comment:
I would make this all upper-case or `MESSAGE` lower-case

Reply:
done